### ISSUE TITLE
behavior with missing avatar images

### ISSUE THREAD

**Issue by ttasovac** [stefanprobst/dariah-campus-mdx#14]
@stefanprobst, here's the situation I've encountered in the editorial branch. One of our resources (Create Impact with your e-Humanities and e-Heritage Research) has multiple authors including juliane.

```
title: Create Impact with your e-Humanities and e-Heritage Research
date: 2019-05-03
authors:
  - ulrike
  - juliane
  - klaus-t 
categories:
```
The corresponding entry in the author yaml file for juliane has an avatar link, but this is an incorrect link because we currently have no image at that url. 

in this situation, `yarn develop` doesn't throw errors, the site gets built, but going to /resources will display
<img width="1061" alt="Скриншот 2019-08-12 19 41 26" src="https://user-images.githubusercontent.com/5304839/62885426-30bd2b00-bd39-11e9-866b-0a51112c202f.png">

Even if we comment out the offending avatar link, i.e. if we have no url there, the resources page will still show the same error. That's probably because we have defined the default avatar image to be used only if the resource has no author named, but we should consider what to do when the author is named, but the avatar image is missing. 

I think the behavior we want is: 
- if the resource has no authors at all, we use the DARIAH-CAMPUS user with the corresponding D-C avatar, as is already the case
- if the resource has a named author, but that author is completely missing from the yaml file, we should ignore that user; that means that if it's the only one, we'll end up with DARIAH-CAMPUS, and if it's one of two or more, it will simply not be displayed. 
- for each author named in the resource that exists in the author yaml file but has an empty or incorrect avatar url, we should use a default person avatar image. That means, we should have an image called avatar.jpg or whatever, with a gender-neutral silhouette of a person. 

I think we need all this just to be on the safe side if, god forbid, we push something like that to production.

At the same time, I wonder if we can use some hooks to throw errors during `yarn develop` so that the repo owner can all those before pushing to production? If that is possible, that we probably don't have to worry about all those conditionals above.

Let me know what you think.

**Comment by stefanprobst** [comment]
thanks, this is super helpful because we're figuring out how this whole thing is going to be used in the wild!

i can look into this more closely on wednesday -- just two quick notes: netlify will never push a build live when there has been a build error, which is :+1:
we should decide if we want to throw an error or use a default fallback image and log a warning message when we encounter a missing avatar

**Comment by stefanprobst** [comment]
> god forbid, we push something like that to production.

this could not happen.

when running `yarn develop`, you actually *don't* want your development server to crash when there is an error -- you'll use `develop` while changing stuff and have a live preview of the changes with hot module reloading

what you push to production is the bundle that gets built with `yarn build`. this will actually fail in the above case when generating the static html with this console message:
```
 ERROR #95313 

Building static HTML failed for path "/resources"

See our docs page for more info on this error: https://gatsby.dev/debug-html


  59 |             key={author.slug}
  60 |             className={styles.authorImage}
> 61 |             fixed={author.avatar.image.fixed}
     |                                  ^
  62 |           />
  63 |         ))}
  64 |         <Link className={styles.readMoreLink} to={path}>


  WebpackError: TypeError: Cannot read property 'image' of null
  
  - PostPreview.js:61 
    src/components/PostPreview/PostPreview.js:61:34
  
  - PostPreview.js:57 PostPreview
    src/components/PostPreview/PostPreview.js:57:30
```

**Comment by ttasovac** [comment]
oh, of course. then we're good —  the most important thing is that the build fails. so, forget about the overcomplicated stuff I wrote above. 

there is however one scenario we should add: the default silhouette of a person for named authors whose pictures we either couldn't find, or those who don't like to have their pictures used. 

the easiest thing from the user perspective would be that the default user icon gets displayed automatically  when a user exist in the yaml file but his or her avatar url field is empty. 

could you set up the avatar preview component to use the fa-user icon in those cases instead of an actual image? 

so to recap:

- if a resource has no named user, the default user is DARIAH-CAMPUS with its own image (D-C logo)
- if a resource has a named user, but no such user exists in the yaml file, throw error on build
- if a resource has a named user, but the avatar url is wrong (pointing to no actual image), throw error on build
- if a resource has a named user, but the avatar url is empty or missing altogether, use the default user icon (preferably fa-user)

**Comment by DeborahThorpe** [comment]
I've just added a few editors for whom there is no photo and did not see this thread until now.  I added a note to remind myself to ask you: is it best to include the 'avatar' field and leave it blank, or just leave 'avatar' out altogether (as I did previously)

**Comment by stefanprobst** [comment]
to Gatsby it makes no difference whether a field is empty or not present, so i think it's best to just leave an empty field out.
i *did* leave all possible fields on the default `dariah` entry so it's clear what fields are possible
